packages/material-ui-lab/src/Masonry/Masonry.js

```diff
@@ -0,0 +1,142 @@
+// A masonry component using the following libs as inspiration.
+//
+// For the implementation:
+// - https://github.com/STRML/react-grid-layout
+// - https://material.angularjs.org/latest/demo/gridList
+// - https://next.material-ui.com/components/image-list/#masonry-image-list
+// - https://github.com/desandro/masonry
+// - https://mdbootstrap.com/docs/react/layout/masonry/
+// - https://muuri.dev/
+// - https://github.com/bigbite/macy.js
+// - https://github.com/paulcollett/react-masonry-css
+// - https://bvaughn.github.io/react-virtualized/#/components/Masonry
+// - https://github.com/jaredLunde/masonic
```

Comment:
Hmm... Although it isn't listed, it seems to follow an approach similar to https://w3bits.com/css-grid-masonry/

This has a couple of limitations, as mentioned: https://github.com/mui-org/material-ui/pull/23954#issuecomment-748319639

The docs page says "Masonry maintains a list of content blocks with a consistent width but variable height. ", but it appears that the height has to be in increments of the gap size, otherwise the gap is expanded to make up the difference:

![image](https://user-images.githubusercontent.com/357702/124971759-8089f700-e021-11eb-9e7d-2904dba2050a.png)

I'm not sure if that's going to fly...

Reply:
I totally agree. I will investigate further for a workaround on this issue.

Reply:
I resolved this in this commit: https://github.com/mui-org/material-ui/pull/27169/commits/00d6780715c090e3681a482446ce16f0c246bb4a.